sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/objects/objects.scala

```diff
@@ -408,16 +439,19 @@ object NewInstance {
       cls: Class[_],
       arguments: Seq[Expression],
       dataType: DataType,
+      initializations: Seq[(String, Seq[Expression])] = Nil,
```

Comment:
@cloud-fan, see here an implementation of the modifications you described [previously](https://github.com/apache/spark/pull/20085#issuecomment-364043282).